torchtitan/utils.py

```diff
@@ -36,6 +37,24 @@ def _warn_overwrite_env(env, val):
     os.environ[env] = val
 
 
+def set_determinism(job_config: JobConfig) -> None:
+    """
+    Set Python, PyTorch, CUDA seeds and cudnn settings for reproducibility
+    Use 1 to avoid determinism
+    """
+    if job_config.training.seed == 1:
+        return
+    # CPU determinism
+    torch.manual_seed(job_config.training.seed)
+    # GPU determinism
+    torch.cuda.manual_seed_all(job_config.training.seed)
+    # set deterministic cudnn algorithms
+    torch.backends.cudnn.deterministic = True
```

Comment:
Will it be better to call `torch.use_deterministic_algorithms(True)`?

Reply:
actually i tried that earlier, and it causes torch.compile to break...so I did not use it.

Reply:
According to https://pytorch.org/docs/stable/notes/randomness.html#cuda-convolution-determinism, this seems to be the right way if we want everything to be deterministic. Should we raise an issue to the compiler team?

Reply:
oh ok, thanks @fegin  - yes I will repro this tomorrow and log the issue with them.

Reply:
I filed the promised issue:
https://github.com/pytorch/pytorch/issues/137160